providers-release-config.yml

```diff
@@ -0,0 +1,45 @@
+project:
+  name: airflow-publish
+  description: "Publish airflow packages to PyPI"
+publisher:
+  name: airflow
+  url: https://dist.apache.org/repos/dist/dev/airflow
+  path: "providers/"
+checks:
+  svn:
+    - id: extension
+      description: "Validate svn package extensions"
+      identifiers:
+        - type: regex
+          pattern: ".*(py3-none-any.whl|py3-none-any.whl.asc|py3-none-any.whl.sha512|tar.gz|tar.gz.asc|tar.gz.sha512)$"
+
+    - id: package_name
+      description: "Validate svn package names"
+      identifiers:
+        - type: regex
+          pattern: ".*(apache_airflow_providers.*)$"
+
+  checksum:
+    - id: checksum
+      description: "Validate check sum with SHA512"
+      algorithm: "sha512"
+
+  signature:
+    - id: signature
+      description: "Validate signatures with GPG of packages"
+      method: gpg
+      keys: "https://dist.apache.org/repos/dist/release/airflow/KEYS"
+
+  publish:
+    id: publish
+    description: "Publish airflow providers packages to PyPI"
+    release-type: "RC_VERSION"
```

Comment:
Should we keep separate workflows for rc and final release?

Here the `release-type` defines the which packages to consider for final version of packages.

When setting `RC_VERSION`, it references the dev SVN folder. 
When setting `PYPI_VERSION`, it uses the release SVN folder and compares it against the dev SVN folder. Only packages that match are published to PyPI. The matching logic is for an additional condition where the release manager may have excluded some packages after the voting process
https://github.com/apache/airflow/issues/41937#issuecomment-2507724501 :)

Reply:
This should work a BIT differently.

1) We will have to modify the release process https://github.com/apache/airflow/blob/main/dev/README_RELEASE_PROVIDER_PACKAGES.md  to also add PYPI packages to SVN. We do not do it currently - because we do not technically need them in SVN, but with this workflow we will need it. I can draft a proposal there and we can discuss it with @eladkal (and try it out in this week's release process).

What I think should happen is that we should have "pypi-rc" sub-folder in providers and `pypi` packages (those with `rc*` in the name should be placed there.

2) Then I think it's best to release RC packages from the the `dev/providers/pypi-rc` and when voting is complete we release "final" packages from "dev/providers" -> NOTE from `dev` not `release` - JUST BEFORE we move them to `release`. This way we will not have to compare anything. This will happen after the packages were removed from "dev/release" by the release manager, and just before we run `svn mv` to move them to release. Which will make it very simple - we upload the same final packages that we move to release. 

Does it make sense? I will draft a PR where I will explain the process to follow on the release management process so that we can see and comment on it.

Reply:
Make sense, if there is no comparison required it makes more easy.

This what i have view i am based on above context.

For RC release the svn directory view, in side pypi-rc directory we will have the following files, and no files exists in inside providers directory?
```
dist/dev/airflow/providers/pypi-rc/
                                   apache_airflow_providers_airbyte-4.0.0rc1-py3-none-any.whl.asc
                                   apache_airflow_providers_airbyte-4.0.0rc1-py3-none-any.whl.sha512
                                   apache_airflow_providers_airbyte-4.0.0rc1-py3-none-any.whl
                                   apache_airflow_providers_airbyte-4.0.0rc1.tar.gz.asc
```                                   
Post RC release, the files will be moved to providers directory and the pypi-rc directory will be empty or removed we dont care about it? only the providers directory files will be released to pypi? the view will be like below?

```
dist/dev/airflow/providers/
                            apache_airflow_providers_airbyte-4.0.0-py3-none-any.whl.asc
                            apache_airflow_providers_airbyte-4.0.0-py3-none-any.whl.sha512
                            apache_airflow_providers_airbyte-4.0.0-py3-none-any.whl
                            apache_airflow_providers_airbyte-4.0.0.tar.gz.asc
```

Everything we operate on the dev folder?

For RC release the validations(checksum, signature, svn) would be performed on `dist/dev/airflow/providers/pypi-rc/`?

Do we need to perform validations(checksum, signature, svn) for final release i.e from `dist/dev/airflow/providers/` ?

Please correct me :)

Reply:
No corrections needed :). This allo looks perfect.